### PR TITLE
chore(post_process_group): Unify pipeline duration measurements

### PR DESCRIPTION
**Pull request by hubertsentry**
We currently measure duration in pipeline function, which can be easy to get forgotten. For example, `process_commits` and `process_replay_link` pipelines don't have duration measurements.

To improve this, I'd like to measure duration in `run_post_process_job` for each pipeline.

### CODE REVIEW

File: src/sentry/tasks/post_process.py [hunk 1]
```diff
@@ -699,7 +699,14 @@ def run_post_process_job(job: PostProcessJob):
 
     for pipeline_step in pipeline:
         try:
-            with sentry_sdk.start_span(op=f"tasks.post_process_group.{pipeline_step.__name__}"):
+            with metrics.timer(
+                "sentry.tasks.post_process.run_post_process_job.pipeline.duration",
```
Comment (vartec) [inline src/sentry/tasks/post_process.py:703]: QQ: isn't `sentry.` prefix automatically prepended?

Comment (hubertsentry) [reply src/sentry/tasks/post_process.py:703]: oh, I didn't realize that. There are few places in this file still using "sentry.tasks.post_process" for `metrics.incr`. I can suppress prefix for sure

**Comment by lobsterkatie** [comment]
So does this replace pipeline-specific timers?

**Comment by hubertsentry** [comment]
Yes, we can remove per pipeline timer once this is merged. I also need to make sure there is no other datadog monitor/dashboard references to pipeline timer/counters.

**Review by vartec — APPROVED** [review]

**Pull request merged**